src/main/java/decodex/recipes/RecipeManager.java

```diff
@@ -58,9 +67,37 @@ public Recipe removeRecipe(String name) throws RecipeManagerException {
      */
     public Recipe getRecipe(String name) throws RecipeManagerException {
         if (!recipeList.containsKey(name)) {
-            throw new RecipeManagerException(RecipeManagerException.RECIPE_NOT_FOUND_MESSAGE);
+            throw new RecipeManagerException(ErrorMessages.RECIPE_NOT_FOUND_MESSAGE);
         }
         return recipeList.get(name);
     }
 
+    public Recipe getEditedRecipe() throws RecipeManagerException {
+        if (editedRecipeName == null) {
+            throw new RecipeManagerException(ErrorMessages.EDITED_RECIPE_NOT_FOUND_MESSAGE);
+        }
+        return getRecipe(editedRecipeName);
+    }
+
+    public void selectRecipe(String name) throws RecipeManagerException {
```

Comment:
Method name is similar to `getRecipe`, which sounds like both methods serve the same purpose on first glance. Consider changing to `setRecipeForEdit` or `setEditingRecipe`, etc.